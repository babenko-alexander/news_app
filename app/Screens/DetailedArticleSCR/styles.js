import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '$primaryBackgroundColor',
    paddingHorizontal: 15
  },
  primaryText: {
    color: '$primaryColor',
    borderColor: '$primaryBorderColor',
    fontSize: '$primaryFontSize',
    textAlign: 'center',
    marginBottom: 15
  },
  secondaryText: {
    color: '$secondaryColor',
    borderColor: '$secondaryBorderColor',
    fontSize: '$secondaryFontSize',
    textAlign: 'right',
    paddingRight: 12,
    marginBottom: 15
  },
  articleText: {
    color: '$secondaryColor',
    borderColor: '$secondaryBorderColor',
    fontSize: 20,
    textAlign: 'center',
    marginBottom: 15 
  }
});

export default styles;
