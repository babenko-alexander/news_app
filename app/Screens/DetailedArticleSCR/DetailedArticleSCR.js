import React, { PureComponent } from 'react';
import {  View, Text, SafeAreaView, Image, Linking } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import styles from './styles';

import SubmitBTN from "../../Components/Buttons/SubmitBTN";

export default class DetailedArticleSCR extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    let { activeArticle } = this.props;
    return (
      <SafeAreaView style={{ flex: 1, justifyContent: 'space-between' }}>
        <KeyboardAwareScrollView>
            <View style={{ alignItems: 'center' }}>
              <Text style={styles.primaryText}> {activeArticle.title} </Text>
              <Image
                id={'article_main_image'} 
                style={{ width: 250, height: 250, marginVertical: 8 }}
                source={{
                  uri: activeArticle.urlToImage
                }}
              />
              <Text style={styles.articleText}> {activeArticle.description} </Text>
              <View style={{ alignSelf: 'flex-end' }}>
              <Text style={styles.secondaryText}> {activeArticle.publishedAt} </Text>
              <Text style={styles.secondaryText}>Автор: {activeArticle.author ? activeArticle.author : 'UNKNOWN'} </Text>
              </View>
            </View>
        </KeyboardAwareScrollView>
        <View style={{ paddingBottom: 7 }}>
        {SubmitBTN("Перейти на сторінку новини", () =>
          Linking.openURL(activeArticle.url)
        )}
        {SubmitBTN("Повернутись до новин", () =>
          this.props.navigation.goBack()
        )}
        </View>
      </SafeAreaView>
    );
  }
}
