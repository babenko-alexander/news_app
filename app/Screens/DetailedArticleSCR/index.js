import { connect } from 'react-redux';
import DetailedArticleSCR from './DetailedArticleSCR';

import { addNewsOK, addNewsERR } from '../../store/reducers/newsArr';

function MSTP(state) {
  return {
    newsArr: state.newsArr,
    activeArticle: state.activeArticle
  };
}

function MDTP(dispatch) {
  return {
    addNewsOK: data => dispatch(addNewsOK(data)),
    addNewsERR: () => dispatch(addNewsERR())
  };
}

export default connect(
  MSTP,
  MDTP
)(DetailedArticleSCR);