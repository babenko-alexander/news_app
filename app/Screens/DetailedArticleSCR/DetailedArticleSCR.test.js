import React from 'react';
import {shallow} from 'enzyme';
import DetailedArticleSCR from './DetailedArticleSCR';

// import configureStore from 'redux-mock-store';
// const middlewares = [ thunk ];
// const mockStore = configureStore();

import { article } from '../../config/jest/mockData';


describe('DetailedArticleSCR component', () => {
  let component;
  beforeEach(()=>{
    // store = mockStore({});
    component = shallow(<DetailedArticleSCR activeArticle={article} />);
    // component = mount(<Provider store={store}><DetailedArticleSCR activeArticle={article} /></Provider> );
  })
  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot("DetailedArticleSCR snapshot");
  });
  it('contains defined image source', () => {
    expect(component.find('Image[id="article_main_image"]').prop('source')).toBeDefined()
  });
  // it('+++ check action on dispatching ', () => {
  //   let action;
  //   store.dispatch(addNewsOK([]));
  //   store.dispatch(addNewsERR());
  //   action = store.getActions();
  //   expect(action[0].type).toBe("FETCH_NEWS_SUCCESS");
  //   expect(action[1].type).toBe("FETCH_NEWS_FAILURE");
  //   });
});
