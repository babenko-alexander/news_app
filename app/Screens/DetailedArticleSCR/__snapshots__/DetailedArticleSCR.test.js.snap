// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedArticleSCR component should match to snapshot: DetailedArticleSCR snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetailedArticleSCR
    activeArticle={
      Object {
        "author": "Тори Путимцева",
        "description": "Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР",
        "id": "2019-07-17T12:54:00Zhttps://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
        "publishedAt": "2019-07-17   12:54:00",
        "title": "ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
        "url": "https://news.google.com/__i/rss/rd/articles/CBMiZmh0dHBzOi8vem5hai51YS9wb2xpdGljcy8yNDczODctZGJyLXBvb2JpY3lhbG8tc2lsb3l1LXByaXZvbG9rdGktcG9yb3NoZW5rYS15YWtzaGNoby1wcm9kb3Z6aGl0LWJpZ2F0adIBamh0dHBzOi8vYW1wLnpuYWoudWEvcG9saXRpY3MvMjQ3Mzg3LWRici1wb29iaWN5YWxvLXNpbG95dS1wcml2b2xva3RpLXBvcm9zaGVua2EteWFrc2hjaG8tcHJvZG92emhpdC1iaWdhdGk?oc=5",
        "urlToImage": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <KeyboardAwareScrollViewMock
          enableAutomaticScroll={true}
          enableOnAndroid={false}
          enableResetScrollToCoords={true}
          extraHeight={75}
          extraScrollHeight={0}
          keyboardOpeningTime={250}
          viewIsInsideTabBar={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Text>
               
              ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua
               
            </Text>
            <Image
              id="article_main_image"
              source={
                Object {
                  "uri": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
                }
              }
              style={
                Object {
                  "height": 250,
                  "marginVertical": 8,
                  "width": 250,
                }
              }
            />
            <Text>
               
              Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР
               
            </Text>
            <View
              style={
                Object {
                  "alignSelf": "flex-end",
                }
              }
            >
              <Text>
                 
                2019-07-17   12:54:00
                 
              </Text>
              <Text>
                Автор: 
                Тори Путимцева
                 
              </Text>
            </View>
          </View>
        </KeyboardAwareScrollViewMock>,
        <View
          style={
            Object {
              "paddingBottom": 7,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.7}
            onPress={[Function]}
            style={
              Object {
                "borderRadius": 50,
              }
            }
          >
            <View>
              <Text>
                Перейти на сторінку новини
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.7}
            onPress={[Function]}
            style={
              Object {
                "borderRadius": 50,
              }
            }
          >
            <View>
              <Text>
                Повернутись до новин
              </Text>
            </View>
          </TouchableOpacity>
        </View>,
      ],
      "style": Object {
        "flex": 1,
        "justifyContent": "space-between",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <Text>
               
              ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua
               
            </Text>
            <Image
              id="article_main_image"
              source={
                Object {
                  "uri": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
                }
              }
              style={
                Object {
                  "height": 250,
                  "marginVertical": 8,
                  "width": 250,
                }
              }
            />
            <Text>
               
              Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР
               
            </Text>
            <View
              style={
                Object {
                  "alignSelf": "flex-end",
                }
              }
            >
              <Text>
                 
                2019-07-17   12:54:00
                 
              </Text>
              <Text>
                Автор: 
                Тори Путимцева
                 
              </Text>
            </View>
          </View>,
          "enableAutomaticScroll": true,
          "enableOnAndroid": false,
          "enableResetScrollToCoords": true,
          "extraHeight": 75,
          "extraScrollHeight": 0,
          "keyboardOpeningTime": 250,
          "viewIsInsideTabBar": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text>
                 
                ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua
                 
              </Text>,
              <Image
                id="article_main_image"
                source={
                  Object {
                    "uri": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
                  }
                }
                style={
                  Object {
                    "height": 250,
                    "marginVertical": 8,
                    "width": 250,
                  }
                }
              />,
              <Text>
                 
                Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР
                 
              </Text>,
              <View
                style={
                  Object {
                    "alignSelf": "flex-end",
                  }
                }
              >
                <Text>
                   
                  2019-07-17   12:54:00
                   
                </Text>
                <Text>
                  Автор: 
                  Тори Путимцева
                   
                </Text>
              </View>,
            ],
            "style": Object {
              "alignItems": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " ",
                  "ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
                  " ",
                ],
                "style": undefined,
              },
              "ref": null,
              "rendered": Array [
                " ",
                "ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
                " ",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "article_main_image",
                "source": Object {
                  "uri": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
                },
                "style": Object {
                  "height": 250,
                  "marginVertical": 8,
                  "width": 250,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " ",
                  "Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР",
                  " ",
                ],
                "style": undefined,
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР",
                " ",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Text>
                     
                    2019-07-17   12:54:00
                     
                  </Text>,
                  <Text>
                    Автор: 
                    Тори Путимцева
                     
                  </Text>,
                ],
                "style": Object {
                  "alignSelf": "flex-end",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      " ",
                      "2019-07-17   12:54:00",
                      " ",
                    ],
                    "style": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "2019-07-17   12:54:00",
                    " ",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "Автор: ",
                      "Тори Путимцева",
                      " ",
                    ],
                    "style": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    "Автор: ",
                    "Тори Путимцева",
                    " ",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TouchableOpacity
              activeOpacity={0.7}
              onPress={[Function]}
              style={
                Object {
                  "borderRadius": 50,
                }
              }
            >
              <View>
                <Text>
                  Перейти на сторінку новини
                </Text>
              </View>
            </TouchableOpacity>,
            <TouchableOpacity
              activeOpacity={0.7}
              onPress={[Function]}
              style={
                Object {
                  "borderRadius": 50,
                }
              }
            >
              <View>
                <Text>
                  Повернутись до новин
                </Text>
              </View>
            </TouchableOpacity>,
          ],
          "style": Object {
            "paddingBottom": 7,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.7,
              "children": <View>
                <Text>
                  Перейти на сторінку новини
                </Text>
              </View>,
              "onPress": [Function],
              "style": Object {
                "borderRadius": 50,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text>
                  Перейти на сторінку новини
                </Text>,
                "style": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Перейти на сторінку новини",
                  "style": undefined,
                },
                "ref": null,
                "rendered": "Перейти на сторінку новини",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.7,
              "children": <View>
                <Text>
                  Повернутись до новин
                </Text>
              </View>,
              "onPress": [Function],
              "style": Object {
                "borderRadius": 50,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text>
                  Повернутись до новин
                </Text>,
                "style": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Повернутись до новин",
                  "style": undefined,
                },
                "ref": null,
                "rendered": "Повернутись до новин",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "SafeAreaView",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <KeyboardAwareScrollViewMock
            enableAutomaticScroll={true}
            enableOnAndroid={false}
            enableResetScrollToCoords={true}
            extraHeight={75}
            extraScrollHeight={0}
            keyboardOpeningTime={250}
            viewIsInsideTabBar={false}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text>
                 
                ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua
                 
              </Text>
              <Image
                id="article_main_image"
                source={
                  Object {
                    "uri": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
                  }
                }
                style={
                  Object {
                    "height": 250,
                    "marginVertical": 8,
                    "width": 250,
                  }
                }
              />
              <Text>
                 
                Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР
                 
              </Text>
              <View
                style={
                  Object {
                    "alignSelf": "flex-end",
                  }
                }
              >
                <Text>
                   
                  2019-07-17   12:54:00
                   
                </Text>
                <Text>
                  Автор: 
                  Тори Путимцева
                   
                </Text>
              </View>
            </View>
          </KeyboardAwareScrollViewMock>,
          <View
            style={
              Object {
                "paddingBottom": 7,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.7}
              onPress={[Function]}
              style={
                Object {
                  "borderRadius": 50,
                }
              }
            >
              <View>
                <Text>
                  Перейти на сторінку новини
                </Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.7}
              onPress={[Function]}
              style={
                Object {
                  "borderRadius": 50,
                }
              }
            >
              <View>
                <Text>
                  Повернутись до новин
                </Text>
              </View>
            </TouchableOpacity>
          </View>,
        ],
        "style": Object {
          "flex": 1,
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text>
                 
                ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua
                 
              </Text>
              <Image
                id="article_main_image"
                source={
                  Object {
                    "uri": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
                  }
                }
                style={
                  Object {
                    "height": 250,
                    "marginVertical": 8,
                    "width": 250,
                  }
                }
              />
              <Text>
                 
                Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР
                 
              </Text>
              <View
                style={
                  Object {
                    "alignSelf": "flex-end",
                  }
                }
              >
                <Text>
                   
                  2019-07-17   12:54:00
                   
                </Text>
                <Text>
                  Автор: 
                  Тори Путимцева
                   
                </Text>
              </View>
            </View>,
            "enableAutomaticScroll": true,
            "enableOnAndroid": false,
            "enableResetScrollToCoords": true,
            "extraHeight": 75,
            "extraScrollHeight": 0,
            "keyboardOpeningTime": 250,
            "viewIsInsideTabBar": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text>
                   
                  ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua
                   
                </Text>,
                <Image
                  id="article_main_image"
                  source={
                    Object {
                      "uri": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
                    }
                  }
                  style={
                    Object {
                      "height": 250,
                      "marginVertical": 8,
                      "width": 250,
                    }
                  }
                />,
                <Text>
                   
                  Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР
                   
                </Text>,
                <View
                  style={
                    Object {
                      "alignSelf": "flex-end",
                    }
                  }
                >
                  <Text>
                     
                    2019-07-17   12:54:00
                     
                  </Text>
                  <Text>
                    Автор: 
                    Тори Путимцева
                     
                  </Text>
                </View>,
              ],
              "style": Object {
                "alignItems": "center",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    " ",
                    "ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
                    " ",
                  ],
                  "style": undefined,
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "ДБР пообіцяло силою приволокти Порошенка, якщо продовжить бігати - Znaj.ua",
                  " ",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "article_main_image",
                  "source": Object {
                    "uri": "https://znaj.ua/images/2019/07/17/nMb3mO2ewMrQ3BD9Fd1NAkV2QdQwJk4v0jGNPZDW.jpeg",
                  },
                  "style": Object {
                    "height": 250,
                    "marginVertical": 8,
                    "width": 250,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    " ",
                    "Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР",
                    " ",
                  ],
                  "style": undefined,
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Петра Порошенка в будь-якому випадку змусять прийти на допит в ГБР",
                  " ",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Text>
                       
                      2019-07-17   12:54:00
                       
                    </Text>,
                    <Text>
                      Автор: 
                      Тори Путимцева
                       
                    </Text>,
                  ],
                  "style": Object {
                    "alignSelf": "flex-end",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        " ",
                        "2019-07-17   12:54:00",
                        " ",
                      ],
                      "style": undefined,
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "2019-07-17   12:54:00",
                      " ",
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        "Автор: ",
                        "Тори Путимцева",
                        " ",
                      ],
                      "style": undefined,
                    },
                    "ref": null,
                    "rendered": Array [
                      "Автор: ",
                      "Тори Путимцева",
                      " ",
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TouchableOpacity
                activeOpacity={0.7}
                onPress={[Function]}
                style={
                  Object {
                    "borderRadius": 50,
                  }
                }
              >
                <View>
                  <Text>
                    Перейти на сторінку новини
                  </Text>
                </View>
              </TouchableOpacity>,
              <TouchableOpacity
                activeOpacity={0.7}
                onPress={[Function]}
                style={
                  Object {
                    "borderRadius": 50,
                  }
                }
              >
                <View>
                  <Text>
                    Повернутись до новин
                  </Text>
                </View>
              </TouchableOpacity>,
            ],
            "style": Object {
              "paddingBottom": 7,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.7,
                "children": <View>
                  <Text>
                    Перейти на сторінку новини
                  </Text>
                </View>,
                "onPress": [Function],
                "style": Object {
                  "borderRadius": 50,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Text>
                    Перейти на сторінку новини
                  </Text>,
                  "style": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Перейти на сторінку новини",
                    "style": undefined,
                  },
                  "ref": null,
                  "rendered": "Перейти на сторінку новини",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.7,
                "children": <View>
                  <Text>
                    Повернутись до новин
                  </Text>
                </View>,
                "onPress": [Function],
                "style": Object {
                  "borderRadius": 50,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Text>
                    Повернутись до новин
                  </Text>,
                  "style": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Повернутись до новин",
                    "style": undefined,
                  },
                  "ref": null,
                  "rendered": "Повернутись до новин",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "SafeAreaView",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
